plaso/parsers/text_plugins/apple_pstxt.py

```diff
@@ -0,0 +1,289 @@
+      event_data.start_time = self._ParseStartTime(start_time_string)
+
+      parser_mediator.ProduceEventData(event_data)
+
+  def _ParseStartTime(self, start_time_string):
+    """Parses the start time element of a log line.
+
+    Args:
+      start_time_string (str): start time element of a log line.
+
+    Returns:
+      dfdatetime.TimeElements: date and time value.
+
+    Raises:
+      ParseError: if a valid date and time value cannot be derived from
+          the time elements.
+    """
+    try:
+      # Case where it's just a time
+      if (
+          start_time_string[0] in pyparsing.nums
+          and start_time_string[-1] == 'M'):
+        parsed_elements = self._TIME.parse_string(start_time_string)
+
+        hours = int(parsed_elements.get('hours'))
+        minutes = int(parsed_elements.get('minutes'))
+        ampm = parsed_elements.get('ampm')
+        if ampm == 'PM':
+          hours += 12
+        if hours == 24:
+          hours = 0
+
+        # Retrieve year, month, day from dateless helper
+        year = self._date[0]
+        month = self._date[1]
+        day = self._date[2]
+
+        time_elements_tuple = (year, month, day, hours, minutes, 0)
+
+      # Case where it is a weekday with a time
+      elif start_time_string[0] in pyparsing.alphas:
+        parsed_elements = self._WEEKDAY.parse_string(start_time_string)
+
+        hours = int(parsed_elements.get('hours'))
+        weekday_string = parsed_elements.get('weekday')
+        weekday = self._GetWeekDayFromString(weekday_string)
+        ampm = parsed_elements.get('ampm')
+        if ampm == 'PM':
+          hours += 12
+        if hours == 24:
+          hours = 0
+
+        # Retrieve year, month, day from dateless helper
+        year = self._date[0]
+        month = self._date[1]
+        day = self._date[2]
+
+        date_from_file = datetime.datetime(year, month, day)
+        days_before = (date_from_file.isoweekday() - weekday + 7) % 7
+        date_of_record = date_from_file - datetime.timedelta(days=days_before)
+
+        time_elements_tuple = (
+            date_of_record.year, date_of_record.month, date_of_record.day,
+            hours, 0, 0)
+
+      # Case where it is a date with no time
+      elif start_time_string[0] in pyparsing.nums:
+        parsed_elements = self._DATE.parse_string(start_time_string)
+
+        year = int(parsed_elements.get('year')) + 2000
+        month_str = parsed_elements.get('month')
+        month = self._GetMonthFromString(month_str)
+        day = int(parsed_elements.get('day'))
+
+        time_elements_tuple = (year, month, day, 0, 0, 0)
+
+      else:
+        raise errors.ParseError('start time was not in an expected format.')
+
+      return dfdatetime_time_elements.TimeElements(
+        time_elements_tuple=time_elements_tuple)
+
+    except (TypeError, ValueError) as exception:
+      raise errors.ParseError(
+        f'Unable to parse time elements with error: {exception}')
```

Comment:
style nit: to keep consistent with the rest of the code base please use `{exception!s}`